src/accessories/TokenArranger.ttslua

```diff
@@ -244,17 +284,70 @@ function layout(_, _, isRightClick)
   local pos           = self.getPosition() + Vector(3.55, -0.05, -3.95)
   local location      = { x = pos.x, y = pos.y, z = pos.z }
   local current_value = data[1].value
+  local token_count = 0
+	local value_count = 1
+	local cumulative_percentage = 0
 
   for _, item in ipairs(data) do
     if item.value ~= current_value then
+      if percentage then
+				cumulative_percentage = cumulative_percentage + math.floor((token_count / #data) * 10000) / 100
+				self.createButton({
+					label = string.format("%s", string.format("%05.2f", math.floor((token_count / #data) * 10000) / 100) .. "%"),
+					click_function = "print",
+					width = 0,
+					height = 0,
+					scale = {2, 2, 2},
+					font_color = {r=1,g=1,b=1},
+					position = (Vector(2.2, -0.05, -2.675) + Vector((0.875 * token_count) + 0.1, 0, 0.875 * value_count))
+				})
```

Comment:
It would be nice if this was turned into a function instead, since this button creation is used 4 times in the script

Reply:
Great suggestion, implemented